app/javascript/mastodon/models/notification_group.ts

```diff
@@ -204,24 +211,37 @@ export function createNotificationGroupFromNotificationJSON(
     case 'mention':
     case 'poll':
     case 'update':
-      return { ...group, statusId: notification.status?.id };
+      return {
+        ...group,
+        type: notification.type,
```

Comment:
I assume those changes were made so Typescript better understands the notification type, but I am not sure I understand what this PR changed that requires it?

Reply:
The is the first commit. The second one does not require it, but as I was working on the PR, I noticed Typescript did not catch type errors there, because of the `as NotificationGroup` which basically bypassed it.

Reply:
This is required for the change in the first commit because Typescript does not keep track of the fact that `group.type` was equal to `notification.type`, while in the `switch` case, it does keep track of the value and can correctly handle the different cases.